### DISCUSSION TITLE
How Next _error page works exactly?

### DISCUSSION THREAD

**Discussion by pablocdev** [vercel/next.js#31302] [Help]
Hi!

I'm trying to implement custom error handling for my Next 12 app, and I've found a question that I couldn't ask by myself.

What I have understood:
- When a 404 error happens, and `getServerSideProps/getStaticProps` returns `{ notFound: true }`, Next loads `pages/404.tsx`
- When a server error happens, for example, throwing an error on `getServerSideProps`, Next loads `pages/500.tsx`
- When a Client Side error happens, Next loads `pages/_error.tsx`

But there is an example in Next's documentation that make me doubt.

![image](https://user-images.githubusercontent.com/23481169/141304118-85b92de0-0df9-451f-8f34-1e1358a39479.png)
[https://nextjs.org/docs/advanced-features/custom-error-page](https://nextjs.org/docs/advanced-features/custom-error-page)

My question is: 

In my project I have created the 404, 500 and _error pages, is there any chance that a *Server Side* error ends up loading the _error page?

Thanks in advance!

**Comment by Yimiprod** [comment]
Is there someone that has any insight about this ?
I currently need to handle error client side to display the error page but i don't have any idea how to redirect to this page, with the correct statusCode.

Atm we use an ErrorBoundary component but i don't know how to correctly display my custom `_error` Page from there.
